### PR TITLE
Add script to make mostly static linux builds.

### PR DESCRIPTION
**Pull request by jcvernaleo**

### CODE REVIEW

File: build_linux.sh [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+# This could definitely be smarter and not always rebuild the deps.
+# You must already have downloaded the ADL sdk from AMD and put the
+# zip file in the same directory as the cgminer repo to use this.
+#
+# 2016/04/25
+# jcv@decred.org
+
+touch build.log
+
+echo "Downloading dependancies if needed."
```
Comment (jolan) [inline build_linux.sh]: dependencies

File: build_linux.sh [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+tar -xzf $ZLIB.tar.gz
+cd $ZLIB
+./configure --static --prefix=$PREF >> build.log 2>&1
+make install >> build.log 2>&1
+cd ..
+
+echo "Building curl."
+tar -xzf $CURL.tar.gz
+cd $CURL
+./configure --disable-rt --disable-rtsp --disable-libcurl-option --disable-ipv6 --disable-verbose --without-ca-bundle --disable-telnet --disable-tftp --disable-pop3 --disable-imap --disable-smb --disable-smtp --disable-gopher --disable-ftp --disable-file --disable-dict --disable-unix-sockets --disable-manual --disable-shared --enable-static --prefix=$PREF >> build.log 2>&1
```
Comment (jolan) [inline build_linux.sh]: For ccminer on Arch Linux, I need --without-libidn --without-librtmp --without-libssh2 --disable-ldap --disable-ldaps

Will test this in a bit to confirm.

Comment (jcvernaleo) [inline build_linux.sh]: I didn't notice any sign of those in my binary so I assumed I didn't need them.  Let me know what your test shows and I can add if needed.

File: build_linux.sh [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+echo "Building ncurses."
+tar -xzf $NCURSES.tar.gz
+cd $NCURSES
+CC=clang Cxx=clang++ ./configure  --enable-pc-files --enable-termcap --without-shared --enable-static --prefix=$PREF --with-pkg-config-libdir=$PREF/lib/pkgconfig --with-terminfo-dirs=/lib/terminfo:/usr/share/terminfo >> build.log 2>&1
+make install >> build.log 2>&1
+cd ..
+
+echo "Getting ADL headers."
+cd ..
+cp ADL_SDK9.zip cgminer/ADL_SDK/
```
Comment (jolan) [inline build_linux.sh]: Might want to check for its existence and gracefully error out, something like:

"ADL headers not found in $PWD; download ADL_SDK9.zip from http://developer.amd.com/tools-and-sdks/graphics-development/display-library-adl-sdk/"

File: build_linux.sh (hunk 2 above)
Comment (jolan) [inline build_linux.sh]: Just confirmed -- they are needed to get a proper static curl on Arch Linux.

Comment (jcvernaleo) [inline build_linux.sh]: I'll check and make sure they don't screw things up on Ubuntu and change if not.

Comment (jcvernaleo) [inline build_linux.sh]: They don't cause an issue on Ubuntu so I added them.

**Comment by jolan** [comment]
OK I can build this just fine on Arch Linux using this script.

**Pull request merged**